sympy/physics/vector/tests/test_point.py

```diff
@@ -126,3 +127,97 @@ def test_point_partial_velocity():
     assert p.partial_velocity(N, u1) == A.x
     assert p.partial_velocity(N, u1, u2) == (A.x, N.y)
     raises(ValueError, lambda: p.partial_velocity(A, u1))
+
+def test_point_vel(): #Basic functionality
+    q1, q2 = dynamicsymbols('q1 q2')
+    N = ReferenceFrame('N')
+    B = ReferenceFrame('B')
+    Q = Point('Q')
+    O = Point('O')
+    Q.set_pos(O, q1 * N.x)
+    raises(ValueError , lambda: Q.vel(N)) #Velocity of Q is not defined
+    O.set_vel(N, q2 * N.y)
+    assert O.vel(N) == q2 * N.y
+    raises(ValueError , lambda : O.vel(B)) #Velocity of O is not defined in B
+
+def test_auto_point_vel():
+    t = dynamicsymbols._t
+    q1, q2 = dynamicsymbols('q1 q2')
+    N = ReferenceFrame('N')
+    B = ReferenceFrame('B')
+    O = Point('O')
+    Q = Point('Q')
+    Q.set_pos(O, q1 * N.x)
+    O.set_vel(N, q2 * N.y)
+    assert Q.vel(N) == q1.diff(t) * N.x + q2 * N.y  # Velocity of Q using O
+    P1 = Point('P1')
+    P1.set_pos(O, q1 * B.x)
+    P2 = Point('P2')
+    P2.set_pos(P1, q2 * B.z)
+    raises(ValueError, lambda : P2.vel(B)) # O's velocity is defined in different frame, and no
+    #point in between has its velocity defined
+    raises(ValueError, lambda: P2.vel(N)) # Pos vector defined in different frame
+
+def test_auto_point_vel_shortest_path():
+    t = dynamicsymbols._t
+    q1, q2 = dynamicsymbols('q1 q2')
+    B = ReferenceFrame('B')
+    P = Point('P')
+    P.set_vel(B, q1 * B.x)
+    P1 = Point('P1')
+    P1.set_pos(P, q2 * B.y)
+    P1.set_vel(B, q1 * B.z)
+    P2 = Point('P2')
+    P2.set_pos(P1, q1 * B.z)
+    P3 = Point('P3')
+    P3.set_pos(P2, 10 * q1 * B.y)
+    assert P3.vel(B) == 10 * q1.diff(t) * B.y + (q1 + q1.diff(t)) * B.z
+
+def test_auto_vel_dont_overwrite():
+    t = dynamicsymbols._t
+    q1, q2 = dynamicsymbols('q1 q2')
+    N = ReferenceFrame('N')
+    P = Point('P1')
+    P.set_vel(N, q1 * N.x)
+    P1 = Point('P1')
+    P1.set_pos(P, q2 * N.y)
+    assert P1.vel(N) == q2.diff(t) * N.y + q1 * N.x
+    assert P.vel(N) == q1 * N.x
+    P1.set_vel(N, q1 * N.z)
+    assert P1.vel(N) == q1 * N.z
+
+def test_auto_point_vel_multiple_frames():
+    t = dynamicsymbols._t
+    q1, q2 = dynamicsymbols('q1 q2')
+    B = ReferenceFrame('B')
+    S = ReferenceFrame('S')
+    N = ReferenceFrame('N')
+    T1 = Point('T1')
+    T1.set_vel(S, q1 * S.x)
+    T2 = Point('T2')
+    T2.set_pos(T1, q2 * S.x)
+    T2.set_vel(N, q1 * N.y)
+    T3 = Point('T3')
+    T3.set_vel(B, q1 * B.z)
+    T3.set_pos(T2, q1 * B.x + q2 * B.y)
+    T3.set_pos(T1, q2 * S.x)
+    T4 = Point('T4')
+    T4.set_pos(T3, q1 * S.z)
+    T5 = Point('T5')
+    T5.set_pos(T4, q2 * S.y)
+    T6 = Point('T6')
+    T6.set_pos(T2, q1 * S.z)
+    assert T6.vel(S) == (q1 + q2.diff(t)) * S.x + q1.diff(t) * S.z
+    T6.set_pos(T3, q1 * B.z)
+    assert T6.vel(B) == (q1 + q1.diff(t)) * B.z
```

Comment:
This test is setup in an invalid, It has two cyclic parts of the graph and is not a tree. SymPy mechanics can only deal with point position trees. You've defined positions from T3 to T1 and T6 that each have two redundant paths. You can make a test with a large tree, but it should be valid.